aws/scenarios/microVMs/microvms/domain.go

```diff
@@ -38,10 +38,12 @@ func generateDomainIdentifier(vcpu, memory int, vmsetName, tag, arch string) str
 	return fmt.Sprintf("ddvm-%s-%s-%s-%d-%d", vmsetName, arch, tag, vcpu, memory)
 }
 func generateNewUnicastMac(ctx *pulumi.Context, domainID string) (pulumi.StringOutput, error) {
-	pulumiRandStr, err := random.NewRandomString(ctx, "random-"+domainID, &random.RandomStringArgs{
-		Length:  pulumi.Int(6),
-		Special: pulumi.Bool(true),
-	})
+	r, err := utils.NewRandom(ctx)
+	if err != nil {
+		return pulumi.StringOutput{}, err
+	}
+
+	pulumiRandStr, err := r.RandomString(domainID, 6, true)
```

Comment:
What is the advantage of using pulumi random generator vs go random ?

Reply:
A MAC address can be any 48-bit identifier. Some bytes have special meanings. In this case we only want to ensure that our mac address is unicast, which is done [here](https://github.com/DataDog/test-infra-definitions/blob/usama.saqib/default-providers/aws/scenarios/microVMs/microvms/domain.go#L54-L56C9) 

Go random would cause pulumi to detect that a mac address is changed on every update, and thus recreate the resource, and its dependencies. 
Pulumi random on the other hand does not create new mac addresses on every update, i.e. random resources are generated only when the stack is created. Therefore, there aren't any spurious resource updates.

Reply:
Ah nice, thank you for the clarification!